### PR TITLE
feat: add testing tier framework (WU-01)

### PR DESCRIPTION
**Pull request by MacAttak**
## Summary

- Adds `specwright-integration-tester` agent for writing integration, contract, and E2E tests at component boundaries — with a strict no-skip-conditions philosophy
- Extends `testing-strategy.md` with tier classification rules mapping TESTING.md boundary types to `[tier: unit/integration/contract/e2e]` annotations
- Extends `sw-plan` to tag ACs with tiers and support both structural (IC-{n}) and behavioral (IC-B{n}) integration criteria
- Updates delegation protocol roster, DESIGN.md, and CLAUDE.md for the new 7-agent architecture

## Acceptance Criteria

| # | Criterion | Evidence | Status |
|---|-----------|----------|--------|
| AC-1 | Integration tester agent definition | `test_integration_tester_agent.py` (43 tests) | PASS |
| AC-2 | Tier Classification section in testing-strategy.md | `test_tier_classification_protocol.py` (22 tests) | PASS |
| AC-3 | Pipeline Flow sw-plan row references [tier: X] | `test_tier_classification_protocol.py::TestAC3` (4 tests) | PASS |
| AC-4 | sw-plan spec writing references tier tagging | `test_sw_plan_tier_tagging.py::TestAC4` (6 tests) | PASS |
| AC-5 | sw-plan integration criteria supports IC-{n} + IC-B{n} | `test_sw_plan_tier_tagging.py::TestAC5` (17 tests) | PASS |
| AC-6 | Delegation roster includes integration-tester | `test_docs_integration_tester.py::TestDelegation*` (11 tests) | PASS |
| AC-7 | DESIGN.md updated (7 agents, tier tagging) | `test_docs_integration_tester.py::TestDesign*` (8 tests) | PASS |
| AC-8 | CLAUDE.md updated with new agent | `test_docs_integration_tester.py::TestClaudeMd*` (2 tests) | PASS |

## Gate Results

| Gate | Status | Findings (B/W/I) |
|------|--------|-------------------|
| Build | PASS | 0/0/0 |
| Tests | PASS | 0/1/2 |
| Security | PASS | 0/0/0 |
| Wiring | PASS | 0/0/1 |
| Semantic | PASS | 0/0/0 |
| Spec | PASS | 0/0/0 |

## Blast Radius

- **NEW**: `core/agents/specwright-integration-tester.md`
- **MODIFIED**: `core/protocols/testing-strategy.md`, `core/protocols/delegation.md`, `core/skills/sw-plan/SKILL.md`, `DESIGN.md`, `CLAUDE.md`, `adapters/claude-code/CLAUDE.md`, `tests/test-claude-code-build.sh`
- **NOT CHANGED**: Existing agents, executor, tester, build system, state protocol, git protocol

## Test plan

- [x] 139 new structural validation tests across 4 test files
- [x] 582 total tests pass (443 existing + 139 new, 0 regressions)
- [x] Build validates on both claude-code and opencode platforms
- [x] CI agent count test updated from 6 → 7

WU-01 of 3 — testing-quality-infrastructure

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: evals/tests/test_sw_plan_tier_tagging.py [hunk 1]
```diff
@@ -0,0 +1,662 @@
...
+        self.assertTrue(
+            has_ic_b,
+            "Integration criteria must define IC-B{n} format for behavioral ICs"
+        )
+
+    def test_ic_b_format_is_checklist_item(self):
+        """IC-B{n} must use the checklist format like IC-{n}."""
+        # Look for: - [ ] IC-B{n}: ...
+        has_checklist = bool(re.search(
+            r"-\s*\[\s*\]\s*IC-B\{?n?\}?\d?:",
```
[the remark this answers is not in the document]
Comment (MacAttak) [reply evals/tests/test_sw_plan_tier_tagging.py]: Good catch. Fixed in 3c1aa6a — regex now uses escaped literal braces `IC-B\{n\}:` instead of the permissive `\{?n?\}?` pattern.

File: core/agents/specwright-integration-tester.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+- **WARN**: gap exists but low-risk
+- **BLOCK**: construct a concrete bypassing implementation; no test catches it
+
+Overall mutation resistance = worst of the three verdicts.
+
+## Output format
+
+- **Test file(s)**: Paths written
+- **Coverage map**: Which AC each test addresses
+- **Edge cases tested**: Boundary/error scenarios
```
[the remark this answers is not in the document]
Comment (MacAttak) [reply core/agents/specwright-integration-tester.md]: Addressed in 3c1aa6a — expensive boundaries now default to `[tier: unit]` (mocked by regular tester). The integration tester's no-skip policy only applies to ACs explicitly tagged with non-unit tiers. External and expensive boundaries stay with the regular tester unless the user reclassifies them as internal in TESTING.md.
